sklearn/tests/test_isotonic.py

```diff
@@ -511,6 +511,24 @@ def test_make_unique_dtype():
         assert_array_equal(x, [2, 3, 5])
 
 
+def test_make_unique_tol():
+    # Check that equality tolerance works
+    x = np.array([0, 1e-16, 1, 1+1e-14])
+    y = x.copy()
+    w = np.ones_like(x)
+    x, y, w = _make_unique(x, y, w)
+    assert_array_equal(x, [0, 1, 1+1e-14])
+
+
+def test_isotonic_inf():
+    # Check that inf values are not returned
+    X = np.array([0., 4.1e-320, 4.4e-314, 1.])
+    y = np.array([0.42, 0.42, 0.44, 0.44])
+    ireg = IsotonicRegression().fit(X, y)
+    y_pred = ireg.predict(np.array([0, 2.1e-319, 5.4e-316, 1e-10]))
+    assert np.all(np.isfinite(y_pred))
+
+
```

Comment:
Also please add a test that checks that the unique filtering log works as expected on a case that is simple enough that we can compute the expected solution intuitively:

```python
>>> from sklearn.isotonic import IsotonicRegression 
>>> X = np.array([0, 1, 1+1e-16, 2], dtype=np.float64)
>>> y = np.array([0, 1, 2, 3], dtype=np.float64)
>>> ireg = IsotonicRegression().fit(X, y)
>>> ireg.predict([0, 0.5, 1, 1.5, 2])
array([0.  , 0.75, 1.5 , 2.25, 3.  ])
```

Here the second and third samples of the training set are collapsed and their corresponding y values are averaged.

```python
>>> ireg.X_thresholds_
array([0., 1., 2.])
>>> ireg.y_thresholds_
array([0. , 1.5, 3. ])
```